MapboxNavigation/NavigationMapView.swift

```diff
@@ -262,6 +266,8 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         addGestureRecognizer(mapTapGesture)
         
         installUserCourseView()
+        
+        styleURL = MGLStyle.navigationDayStyleURL
```

Comment:
This overrides the style URL explicitly passed into `init(frame:styleURL:)` and possibly also `init(coder:)`. I think we should move it into `init(frame:)` and find a way to only conditionally set it in `init(coder:)`. This might be why we hadn’t customized NavigationMapView’s default style URL up to now.

Reply:
Good catch. There is one really strange issue with new approach though. When I add `let styleURL = decoder.decodeObject(of: [URL.self], forKey: "styleURL") as? URL` anywhere in the code in `NavigationMapView` Xcode gets `Segmentation fault: 11` error during build. I think it'd be better to create separate ticket to handle this, because of mentioned issue and to prevent any side effects.

Reply:
URL is a pure Swift type, so it’s incompatible with `NSCoder.decodeObject(of:forKey:)`. But it should work with `TopLevelDecoder.decode(_:from:)`.

Reply:
Interesting finding, on the other hand seems that `TopLevelDecoder.decode(_:from:)` is still part of Combine. In any case I'll consider this approach as well.

Reply:
Oops, I meant `KeyedDecodingContainer.decode(_:)`. We need to create a local decoding container in order to decode that key.